### DISCUSSION TITLE
Option to keep previous modals mounted but hidden, when opening multiple modals

### DISCUSSION THREAD

**Discussion by brumm** [mantinedev/mantine#5841] [Feature requests]
Hey,

I'm displaying a form in a modal. The main action is 'review and confirm' and opens another modal where the form inputs are summarized and the user can confirm the action.

If the user does not want to confirm but go back instead, the form values are lost because the form got unmounted when the next modal opened.

I'd suggest a new option to opt-in to the new behavior, something like `<ModalProvider keepModalsMounted />`.

If you're interested in this, I'd volunteer to implement it 🙌 

Related discussion: https://github.com/orgs/mantinedev/discussions/3063, I opened a new one because I'm offering to build this :)

**Comment by brumm** [comment]
@rtivital Would appreciate a 👍 / 👎 here

**Reply by rtivital** [reply]
I'm not really interested in customizing modals manager behavior, it may produce unexpected bugs that I do not have capacity to fix.
